src/test/agent/test_systemd.py

```diff
@@ -1,69 +1,85 @@
 
         self.systemd = SystemdUnits(self.manager)
 
     def test_start_units(self):
-        self.systemd._get_units = MagicMock(return_value=[self.unit])
+        self.systemd._get_units = MagicMock(return_value=[self.unit1])
 
         self.systemd.start_units(self.manifest)
-        self.unit.start.assert_called_with('replace')
+        self.unit1.start.assert_called_with('replace')
 
     def test_start_units_error(self):
-        self.systemd._get_units = MagicMock(return_value=[self.unit])
-        self.unit.start.side_effect = Exception()
+        self.systemd._get_units = MagicMock(return_value=[self.unit1])
+        self.unit1.start.side_effect = Exception()
 
         self.systemd.start_units(self.manifest)
 
     def test_start_units_already_running(self):
-        self.systemd._get_units = MagicMock(return_value=[self.unit])
-        self.unit.properties.ActiveState = 'active'
+        self.systemd._get_units = MagicMock(return_value=[self.unit1])
+        self.unit1.properties.ActiveState = 'active'
 
         self.systemd.start_units(self.manifest)
-        self.unit.start.assert_not_called()
+        self.unit1.start.assert_not_called()
 
     def test_stop_units(self):
-        self.systemd._get_units = MagicMock(return_value=[self.unit])
+        self.systemd._get_units = MagicMock(return_value=[self.unit1])
         self.systemd.stop_units(self.manifest)
-        self.unit.stop.assert_called_with('replace')
+        self.unit1.stop.assert_called_with('replace')
 
     def test_stop_units_error(self):
-        self.systemd._get_units = MagicMock(return_value=[self.unit])
-        self.unit.stop.side_effect = Exception()
+        self.systemd._get_units = MagicMock(return_value=[self.unit1])
+        self.unit1.stop.side_effect = Exception()
         self.systemd.stop_units(self.manifest)
 
     def test_get_units_loaded(self):
-        self.manager.list_units.return_value = [self.unit]
+        self.manager.list_units.return_value = [self.unit1, self.unit2,
+                                                self.unit3]
 
         units = [u for u in self.systemd._get_units(self.manifest)]
-        self.assertEquals(1, len(units))
-        self.assertEquals(self.unit, units[0])
+        self.assertEquals(3, len(units))
+        self.assertEquals(self.unit1, units[0])
+        self.assertEquals(self.unit2, units[1])
+        self.assertEquals(self.unit3, units[2])
 
     def test_get_units_load(self):
         self.manager.list_units.return_value = []
-        self.manager.load_unit.return_value = self.unit
+        self.manager.load_unit.side_effect = [self.unit1, self.unit2,
+                                              self.unit3]
 
         units = [u for u in self.systemd._get_units(self.manifest)]
-        self.assertEquals(1, len(units))
-        self.assertEquals(self.unit, units[0])
+        self.assertEquals(2, len(units))
+        self.assertEquals(self.unit1, units[0])
+        self.assertEquals(self.unit2, units[1])
```

Comment:
@thepwagner was hoping to catch some feedback here:
I don't understand why it's returning `self.unit2` (the 'skipped' unit) but does not return the `bar.timer` unit..